examples/__tests__/cognito.test.ts

```diff
@@ -0,0 +1,503 @@
+        await expect(
+          cognitoUser({
+            username: 'petlejsaj@lirabzoz.fk',
+            userPoolId,
+            retryPolicy: {
+              retries: 0,
+            },
+          }),
+        ).not.toExist();
+      } catch (e) {
+        expect(e).toMatchInlineSnapshot(
+          `[Error: Expected "us-east-1_2KFDP3x2n" user pool not to have a user with username "petlejsaj@lirabzoz.fk"]`,
+        );
+      }
+    });
+  });
+
+  describe('toExistAndMatchObject', () => {
+    it('should succeed when user exists and matches the expected object', async () => {
+      await expect(
+        cognitoUser({
+          username: existingUser,
+          userPoolId,
+        }),
+      ).toExistAndMatchObject({
+        Username: existingUser,
+        Enabled: true,
+        UserAttributes: expect.arrayContaining([
+          {
+            Name: 'email',
+            Value: existingUser,
+          },
+        ]),
+      });
+    });
+
+    it("should fail when user exists but doesn't match the expected object", async () => {
+      try {
+        await expect(
+          cognitoUser({
+            username: existingUser,
+            userPoolId,
+          }),
+        ).toExistAndMatchObject({
+          Username: existingUser,
+          Enabled: false,
+          UserAttributes: expect.arrayContaining([
+            {
+              Name: 'email',
+              Value: 'fake@fake.com',
+            },
+          ]),
+        });
+      } catch (e) {
+        expect(e).toHaveProperty('message');
+        expect(e).toHaveProperty('matcherResult.message');
+        expect(e).toHaveProperty('matcherResult.actual');
+        expect(e).toHaveProperty('matcherResult.expected', {
+          Username: existingUser,
+          Enabled: false,
+          UserAttributes: expect.arrayContaining([
+            {
+              Name: 'email',
+              Value: 'fake@fake.com',
+            },
+          ]),
+        });
+      }
+    });
+
+    it('should fail when user does not exists', async () => {
+      try {
+        const randomEmail = 'ci@nen.ma';
+        await expect(
+          cognitoUser({
+            username: randomEmail,
+            userPoolId,
+          }),
+        ).toExistAndMatchObject({
+          Username: randomEmail,
+          Enabled: true,
+          UserAttributes: expect.arrayContaining([
+            {
+              Name: 'email',
+              Value: existingUser,
+            },
+          ]),
+        });
+      } catch (e) {
+        expect(e).toMatchInlineSnapshot(
+          `[Error: Expected "us-east-1_2KFDP3x2n" user pool to have a user with username "ci@nen.ma"]`,
+        );
+      }
+
+      try {
+        await expect(
+          cognitoUser({
+            username: existingUser,
+            userPoolId,
+            clientConfig: {
+              region: 'us-east-2',
+            },
+          }),
+        ).toExistAndMatchObject({
+          id: '456',
+          title: 'Play Fifa',
```

Comment:
not a big deal but this looks inaccurate

Reply:
fixed it, https://github.com/serverless-guru/sls-jest/pull/73/commits/432083fcb677d2620addfd211af1bc1649e43657